internal/clients/manager.go

```diff
@@ -47,6 +52,12 @@ type AdminAPIClientsManager struct {
 	// endpoints list that should be used for configuring the dataplane.
 	discoveredAdminAPIsNotifyChan    chan []adminapi.DiscoveredAdminAPI
 	gatewayClientsChangesSubscribers []chan struct{}
+	// adminAPIServiceName is the name of admin API service.
+	// When Kong gateway is running with DB, the manager will return the address of admin API service
+	// since KIC need only to send the requests to the admin API once with DB backed Kong gateway.
+	adminAPIServiceName k8stypes.NamespacedName
+	adminAPIServicePort int
+	dbMode              string
```

Comment:
I think we could define an enum for DB mode to not pass a raw string everywhere, e.g. 
```go
type KongDBMode string

const (
	KongDBModeOff      DBMode = "off"
	KongDBModePostgres DBMode = "postgres"
)
```
WDYT?

Reply:
And it can be reused in `test/internal/testenv/testenv.go` where similar enum has been already defined

Reply:
This could be done in another PR (maybe along with definition of router flavors) because this is used in many places.